### ISSUE TITLE
ConnectAsync exception when trying to connect through IP provided by an A record

### ISSUE THREAD

**Issue by GuidoRusso** [jstedfast/MailKit#1165]
Hi, I'm trying to establish a connection with a SMTP server. Normally, we use the exchange provided by the MX record. 
But, I'm having problems when trying to connect with the Address of A (or AAAA in case of IPv6) record.

I was reviewing in the code and noted that internally, when you try to connect with an exchange, a DNS lookup is made to obtain the A record. 

My question is: **Is it possible to establish a connection with a SMTP server by using Address provided by A or AAAA record?** I mean...

                    await smtpClient.ConnectAsync(
                        "216.58.194.206",
                        25,
                        SecureSocketOptions.Auto,
                        new CancellationTokenSource(timeout).Token);

Take into account that Ip pasted in the code is from an A record of domain "google.com".
I got a really generic exception, "A task was canceled".

Thanks.

**Comment by jstedfast** [comment]
You are getting the timeout exception because of your `new CancellationTokenSource(timeout).Token`

My guess is that the timeout is really short.

FWIW, all of the clients use the following method to connect a socket: https://github.com/jstedfast/MailKit/blob/master/MailKit/Net/SocketUtils.cs#L38

I just wrote a test program to verify that Dns.GetHostNameAddressesAsync() correctly handles "216.58.194.206" and it does.

**Comment by GuidoRusso** [comment]
Great, my problem was resolved! Thanks.
Regarding that, Mailkit supports to connect through and AAAA record? I was trying to connect to an SMTP server by using its IPv6 address, but I got an error. 

Watching the code, I saw that you are validating that IPv6 could be mapped to an IPv4. But, I think that is not the same. 

For example, I made a DNS lookup to "gmail.com". I took the lowest preference MX record, and take its exchange. Now I made a DNS lookup to the exchange server, and it returns A and AAAA records.

Establish a connection using IP Addresses from A record works properly.
Establish a connection using IP Address from AAAA record ends up in an error. IPv6 addresses are the following:
* 2607:f8b0:4023:c03::1b
* 2607:f8b0:4023:c03::1a

This tool can connect to an SMTP server successfully by using gmail.com domain (with its IPv6 address)
